ament_tools/build_types/cmake.py

```diff
@@ -368,37 +443,99 @@ def _common_cmake_on_install(self, context):
                 yield BuildAction(
                     prefix + [
                         MSBUILD_EXECUTABLE,
-                        '/p:Configuration=' + self._get_visual_studio_configuration(context),
+                        '/p:Configuration=' + self._get_configuration_from_cmake(context),
                         install_project_file])
             else:
                 self.warn("Could not find Visual Studio project file 'INSTALL.vcxproj'")
+        elif IS_MACOSX:
+            if self._using_xcode_generator(context):
+                # The Xcode CMake generator will produce a file named
+                # install_postBuildPhase.makeRelease in the CMakeScripts directory if there is an
+                # install command in the CMakeLists.txt file of the package. We use that to only
+                # call xcodebuild's install target if there is anything to install
+                install_cmake_file_path = os.path.join(
+                    context.build_space, 'CMakeScripts', 'install_postBuildPhase.makeRelease')
+                install_cmake_file = os.path.isfile(install_cmake_file_path)
+                if install_cmake_file:
+                    if XCODEBUILD_EXECUTABLE is None:
+                        raise VerbExecutionError("Could not find 'xcodebuild' executable")
+                    cmd = prefix + [XCODEBUILD_EXECUTABLE]
+                    cmd += ['-target', 'install']
+                    xcodebuild_flags = []
+                    for flag in context.make_flags:
+                        if not flag.startswith('-l'):
+                            if flag.startswith('-j'):
+                                xcodebuild_flags.append(
+                                    flag.replace(
+                                        '-j',
+                                        '-IDEBuildOperationMaxNumberOfConcurrentCompileTasks='))
+                            else:
+                                xcodebuild_flags.append(flag)
+                    xcodebuild_flags += ['-configuration']
+                    xcodebuild_flags += [self._get_configuration_from_cmake(context)]
+                    cmd.extend(xcodebuild_flags)
```

Comment:
Agreed, I've added @dirk-thomas 's code and moved it to a separate function in https://github.com/ament/ament_tools/pull/127/commits/9ace6e8203c2e333ce37b618935073e15a578bad

Thanks!